src/python/turicreate/config/__init__.py

```diff
@@ -250,6 +250,9 @@ def set_runtime_config(name, value):
     - *TURI_S3_ENDPOINT*: The S3 Endpoint to connect to. If not specified AWS
       S3 is assumed.
 
+    - *TURI_S3_REGION*: The S3 Region to connect to. If not specified AWS
```

Comment:
> `If not specified AWS S3 region is assumed.`

I am not sure this is correct -- specifically, there is more than one region in AWS, and I thought we try all of them sequentially (perhaps starting with the default, `us-east-1`). Though I can't find the code to do that -- the only code I can find seems to default to (and only try) `us-east-1`. @ylow, can you confirm which is the intended behavior here?

Even if we are only defaulting to a single region, we should specify which one it will default to (`us-east-1`, I think) instead of leaving it ambiguous here.

Reply:
first to look up regions through end point
```
const std::map<std::string, std::string> AWS_S3_ENDPOINT_TO_REGION {
  {"s3.amazonaws.com","us-east-1"},
  {"s3-us-west-1.amazonaws.com","us-west-1"},
  {"s3-us-west-2.amazonaws.com","us-west-2"},
  {"s3-eu-west-1.amazonaws.com","eu-west-1"},
  {"s3-eu-central-1.amazonaws.com","eu-central-1"},
  {"s3-ap-southeast-1.amazonaws.com","ap-southeast-1"},
  {"s3-ap-southeast-2.amazonaws.com","ap-southeast-2"},
  {"s3-ap-northeast-1.amazonaws.com","ap-northeast-1"},
  {"s3-ap-northeast-2.amazonaws.com","ap-northeast-2"},
  {"s3-sa-east-1.amazonaws.com","sa-east-1"},
  {"s3-ap-south-1.amazonaws.com","ap-south-1"}}
```
if no association is found, will let `AWS` decide from the endpoint the user provide. On my end is us-west-1 if not specified.

Reply:
I see -- from this, it does look to me like the default is `us-east-1`, since most users would have `s3.amazonaws.com` as their endpoint. Or if the user doesn't set endpoint explicitly, does it bypass this lookup?

Reply:
nah. I'm using a private cloud endpoint which is not in the list above. The region for s3 client is set to empty string. I happen to know the value is `us-west-1` by reading the response from aws afterwards.

Reply:
I see. In that case, we can probably say something like  "If not specified, the region is determined from the endpoint URL, and/or determined automatically on the server." in the doc string.

Reply:
agreed. I shouldn't be this lazy next time. Thanks for poking into this pr.

Reply:
IMO, the workflow should be something like this: 

If TURI_S3_REGION is set, then use that.  Error out if it doesn't work. 
If TURI_S3_REGION is not set, then check the AWS_S3_REGION variable.  If set, then use that.  
If neither TURI_S3_REGION nor AWS_S3_REGION is set, then rely on the aws-cli library to make the decision.  I don't think it makes sense to hardcode aws regions if there's a way to get aws-cli to make that decision for us.

Reply:
Good point! I will check what options aws client offers as we discussed yesterday.

Reply:
Sounds great!  Thanks for working on this!

Reply:
Now AWS_DEDAULT_REGION will be loaded. I think we are good to go.